addons/knobs/src/components/types/Select.js

```diff
@@ -3,26 +3,61 @@ import PropTypes from 'prop-types';
 
 import { Select } from '@storybook/components';
 
+const safeToString = value => (value.toString && value.toString()) || `${value}`;
+
+const stringifyKeys = object =>
+  Object.entries(object).reduce(
+    (finalObject, [key, value]) => ({
+      ...finalObject,
+      [safeToString(key)]: value,
+    }),
+    {}
+  );
+
+const arrayToObject = (finalObject, value) => ({
+  ...finalObject,
+  [safeToString(value)]: value,
+});
+
 class SelectType extends Component {
-  renderOptionList({ options }) {
-    if (Array.isArray(options)) {
-      return options.map(val => this.renderOption(val, val));
-    }
-    return Object.keys(options).map(key => this.renderOption(key, options[key]));
+  static getDerivedStateFromProps({ knob }) {
```

Comment:
I think we can instead use memoization here to achieve the same performance.

What do you think? It's usually recommended to use memoization over getDerivedStateFromProps.